### PR TITLE
Fix useEffect() in sky nav

### PR DESCRIPTION
**Pull request by calebeby**
Closes https://github.com/cloudfour/cloudfour.com-patterns/issues/1909

The useEffect was being called multiple times, where it was before. I don't think it should have been placed where it was, because that is not in the top-level of a component, which is the only place where hooks are allowed.

It seems like Storybook made a change where story components now get rendered multiple times. Because of our placement of useEffect, the sky nav JS gets initialized twice. This caused it to expand and retract immediately when you click the menu button.

I changed the useEffect to be called in a decorator which I think is [actually a component](https://storybook.js.org/tutorials/create-an-addon/react/en/decorators/), so it should be OK to use useEffect there.

This problem exists in other components too. I can go through them and fix them if there are any others where the multiple-initialization of JS is causing a problem, but I figure that is not a launch blocker since for most of them it seems to not matter.

@tylersticka the sky nav JS in the prototypes that I added in https://github.com/cloudfour/cloudfour.com-patterns/pull/1503 is broken also, but I think for a different reason. I didn't look into fixing that since I'm not sure if it's important that those continue to work since we've implemented the production version now. Let me know if you want me to fix those!

https://deploy-preview-1918--cloudfour-patterns.netlify.app/?path=/story/components-sky-nav--dark

### CODE REVIEW

File: src/components/sky-nav/sky-nav.stories.mdx [hunk 1]
```diff
@@ -35,6 +35,17 @@ The Sky Nav is intended for use with our [our dark theme](/docs/design-themes--d
...
+      (story) => {
+        useEffect(() => {
+          const { destroy } = initSkyNav(
+            document.querySelector('.js-sky-nav-menu-toggle')
+          );
+          return destroy;
+        }, []);
+        return story();
+      },
+    ]}
```
[the remark this answers is not in the document]
Comment (calebeby) [reply src/components/sky-nav/sky-nav.stories.mdx]: Yep, thanks!

**Comment by tylersticka** [comment]
> This problem exists in other components too. I can go through them and fix them if there are any others where the multiple-initialization of JS is causing a problem, but I figure that is not a launch blocker since for most of them it seems to not matter.

Can you create an issue for the broader problem?

> the sky nav JS in the prototypes that I added in #1503 is broken also, but I think for a different reason. I didn't look into fixing that since I'm not sure if it's important that those continue to work since we've implemented the production version now. Let me know if you want me to fix those!

It's not important for those to continue to work. Maybe we should remove them if we know they're broken?

**Comment by calebeby** [comment]
> Can you create an issue for the broader problem?

✅ https://github.com/cloudfour/cloudfour.com-patterns/issues/1919

> It's not important for those to continue to work. Maybe we should remove them if we know they're broken?

Remove all the prototypes which have the sky nav? Or remove the sky nav JS from those? I'm fine with either. Should that be in this PR or another?

**Comment by tylersticka** [comment]
> Remove all the prototypes which have the sky nav? Or remove the sky nav JS from those? I'm fine with either. Should that be in this PR or another?

Either one. I have no preference.

**Review by tylersticka — APPROVED** [review]

**Pull request merged**